src/OpenApi/src/Transformers/ActivatedOpenApiDocumentTransformer.cs

```diff
@@ -0,0 +1,24 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.OpenApi.Models;
+
+namespace Microsoft.AspNetCore.OpenApi;
+
+internal sealed class ActivatedOpenApiDocumentTransformer(Type transformerType) : IOpenApiDocumentTransformer
+{
+    internal IOpenApiDocumentTransformer? Transformer { get; set; }
+
+    internal void Initialize(IServiceProvider serviceProvider)
+    {
+        Transformer ??= ActivatorUtilities.CreateInstance(serviceProvider, transformerType) as IOpenApiDocumentTransformer;
+    }
+
+    public Task TransformAsync(OpenApiDocument document, OpenApiDocumentTransformerContext context, CancellationToken cancellationToken)
+    {
+        Debug.Assert(Transformer != null, "Transformer should have been initialized.");
```

Comment:
Why is `Initialize` a separate method?  Is `TransformAsync` expected to be called much more often?

Reply:
Since we use null-coalescing to set the `Transformer` we could initialize it here but I liked the clarity that separating it out into separate calls provided. The fact that this transformer type needs to do some additional work at invocation to activate dependencies from the DI container felt special enough to call out in its own method.

Reply:
I don't get it.  Is it important that that happen after construction but before `TransformAsync`?

Reply:
Actually, following @BrennanConroy's feedback [here](https://github.com/dotnet/aspnetcore/pull/54935#discussion_r1552579784), the `Initialize` method isn't the right way to go since we want scoped/transient services to be resolved again for each request/invocation to the document so this needs to be worked. Done in [`73fc68a` (#54935)](https://github.com/dotnet/aspnetcore/pull/54935/commits/73fc68a58d1a8128004208eb86b9c79cc2b0d704).

Reply:
Do we care about disposable transient transformers?
`class MyTransformer : IDisposable, IOpenApiDocumentTransformer`